core/setup.py

```diff
@@ -0,0 +1,58 @@
+import os
+
+from setuptools import setup
+from setuptools import find_packages
+
+
+PACKAGE_ROOT = os.path.abspath(os.path.dirname(__file__))
+
+with open(os.path.join(PACKAGE_ROOT, 'README.rst')) as file_obj:
+    README = file_obj.read()
+
+# NOTE: This is duplicated throughout and we should try to
+#       consolidate.
+SETUP_BASE = {
+    'author': 'Google Cloud Platform',
+    'author_email': 'jjg+google-cloud-python@google.com',
+    'scripts': [],
+    'url': 'https://github.com/GoogleCloudPlatform/google-cloud-python',
+    'license': 'Apache 2.0',
+    'platforms': 'Posix; MacOS X; Windows',
+    'include_package_data': True,
+    'zip_safe': False,
+    'classifiers': [
+        'Development Status :: 4 - Beta',
+        'Intended Audience :: Developers',
+        'License :: OSI Approved :: Apache Software License',
+        'Operating System :: OS Independent',
+        'Programming Language :: Python :: 2',
+        'Programming Language :: Python :: 2.7',
+        'Programming Language :: Python :: 3',
+        'Programming Language :: Python :: 3.4',
+        'Programming Language :: Python :: 3.5',
+        'Topic :: Internet',
+    ],
+}
+
+
+REQUIREMENTS = [
+    'httplib2 >= 0.9.1',
+    'googleapis-common-protos',
+    'oauth2client >= 2.0.1',
```

Comment:
`< 3.0.0` makes the actual 3.0.0 release the excluded upper bound. `< 3.0.0dev` prevents any 3.0.x pre-release from satisfying the constraint.  In general, the second form would be preferrable.